docs/security.rst

```diff
@@ -0,0 +1,288 @@
+
+.. somewhat surprisingly the "bash" highlighter gives nice results with
+   the pseudo-code notation used in the "Encryption" section.
+
+.. highlight:: bash
+
+========
+Security
+========
+
+Cryptography in Borg
+====================
+
+Attack model
+------------
+
+The attack model of Borg is that the environment of the client process
+(eg. ``borg create``) is trusted and the repository (server) is not. The
+attacker has any and all access to the repository, including interactive
+manipulation (man-in-the-middle) for remote repositories.
+
+Furthermore the client environment is assumed to be persistent across
+attacks (practically this means that the security database cannot be
+deleted between attacks).
+
+Under these circumstances Borg guarantees that the attacker cannot
+
+1. modify the data of any archive without the client detecting the change
+2. rename, remove or add an archive without the client detecting the change
+3. recover plain-text data
+4. recover definite (heuristics based on access patterns are possible)
+   structural information such as the object graph (which archives
+   refer to what chunks)
+
+The attacker can always impose a denial of service per definition (he could
+forbid connections to the repository, or delete it entirely). The attacker
+can gauge how many archives exist in the repository due to the access pattern
+associated with creating archives, and because the manifest size gives this away
+(assuming approximately constant and known length of archive names).
```

Comment:
why should we assume the attacker knows the length of archive names?

more than wild guesses (like: borg user maybe using something similar to the example script from the docs) are not possible here, imho.

assuming this here could leave the wrong impression that the attacker could somehow know them.